unidist/core/backends/mpi/core/controller/api.py

```diff
@@ -181,8 +182,11 @@ def shutdown():
     mpi_state = communication.MPIState.get_instance()
     # Send shutdown commands to all ranks
     for rank_id in range(communication.MPIRank.MONITOR, mpi_state.world_size):
+        # a blocking send is used because completion is necessary for processing to continue
```

Comment:
```suggestion
        # We use a blocking send here because we have to wait for
        # completion of the communication, which is necessary for the pipeline to continue.
```